tests/integration/targets/aci_cloud_provider/tasks/main.yml

```diff
@@ -0,0 +1,51 @@
+# Test code for the ACI modules
+# Copyright: (c) 2020, Lionel Hercot (@lhercot) <lhercot@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+# CLEAN ENVIRONMENT
+- name: Set vars
+  set_fact: 
+    mso_info: &mso_info
+      host: '{{ aci_hostname }}'
+      username: '{{ aci_username }}'
+      password: '{{ aci_password }}'
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("info") }}'
+
+- name: Query system information
+  aci_system:
+    <<: *mso_info
+    id: 1
+    state: query
+  register: version
+
+# QUERY OBJECTS
+- name: Query cloud provider object
+  aci_cloud_provider:
+    <<: *mso_info
+    state: query
+  register: query_cloud
```

Comment:
not sure whether the 'query_cloud' will be registered if the version < 4.1.
if it isn't registered, can we do the following tasks?

Reply:
It seems to pass the not changed check so will let it if it pass.